client/starwhale/utils/config.py

```diff
@@ -248,3 +257,42 @@ def update_instance(
         )
 
         update_swcli_config(**self._config)
+
+
+@click.group(
+    "config",
+    cls=AliasedGroup,
+    help="Configuration management, edit is supported now",
+)
+def config_cmd() -> None:
+    pass
+
+
+@config_cmd.command("edit", aliases=["e"], help="edit the configuration of swlci")
+def __edit() -> None:
+    _edit()
+
+
+def _edit() -> None:
+    _editor = ""
+    if platform == "linux" or platform == "linux2":
+        _editor = os.environ.get(SWEnv.sw_editor, "")
+        if not _editor:
+            _editor = os.environ.get("EDITOR", "vi")
+    elif platform == "darwin" or platform == "win32":
+        _editor = os.environ.get(SWEnv.sw_editor, "")
+        if not _editor:
+            raise NoSupportError(f"No {SWEnv.sw_editor} is configured")
+    else:
+        raise NoSupportError(f"Current platform is not supported by: {platform}")
```

Comment:
```
    if platform == "linux" or platform == "linux2":
        _editor = os.environ.get(SWEnv.sw_editor, "")
        if not _editor:
            _editor = os.environ.get("EDITOR", "vi")
    elif platform == "darwin" or platform == "win32":
        _editor = os.environ.get(SWEnv.sw_editor, "")
        if not _editor:
            raise NoSupportError(f"No {SWEnv.sw_editor} is configured")
    else:
        raise NoSupportError(f"Current platform is not supported by: {platform}")
```
-->
```
from shutil import which
_editor = os.environ.get(SWEnv.sw_editor) or os.environ.get("EDITOR") or "vi"
if which(_editor) is None:
   raise noSupportError(f"no found {_editor} bin in {platform}")
```
?

Reply:
sometime the editor command may be a compose of executable and some options, like:
`code -w` in windows or `/Applications/Sublime.app/Contents/SharedSupport/bin/subl -w` in macos. 
In such cases, `which` may not working as expected. However, the code you provide is really concise.

Reply:
> ```
>     if platform == "linux" or platform == "linux2":
>         _editor = os.environ.get(SWEnv.sw_editor, "")
>         if not _editor:
>             _editor = os.environ.get("EDITOR", "vi")
>     elif platform == "darwin" or platform == "win32":
>         _editor = os.environ.get(SWEnv.sw_editor, "")
>         if not _editor:
>             raise NoSupportError(f"No {SWEnv.sw_editor} is configured")
>     else:
>         raise NoSupportError(f"Current platform is not supported by: {platform}")
> ```
> 
> -->
> 
> ```
> from shutil import which
> _editor = os.environ.get(SWEnv.sw_editor) or os.environ.get("EDITOR") or "vi"
> if which(_editor) is None:
>    raise noSupportError(f"no found {_editor} bin in {platform}")
> ```
> 
> ?

done